x-pack/plugins/apm/public/components/app/TransactionDetails/Transaction/WaterfallContainer/Waterfall/waterfall_helpers/waterfall_helpers.ts

```diff
@@ -231,8 +235,9 @@ function createGetTransactionById(itemsById: IWaterfallIndex) {
 }
 
 export function getWaterfall(
-  hits: Array<Span | Transaction>,
-  entryTransaction: Transaction
+  hits: TraceAPIResponse['trace'],
+  entryTransaction: Transaction,
```

Comment:
When you make the other change we talked about, we will no longer have this `entryTransaction`. Instead we'll have a `entryTransactionId`